plugins/modules/snap.py

```diff
@@ -69,18 +72,33 @@
             - Set options with pattern C(key=value) or C(snap:key=value). If a snap name is given, the option will be applied
               to that snap only. If the snap name is omitted, the options will be applied to all snaps listed in O(name). Options will
               only be applied to active snaps.
+            - Options will only be applied when C(state) is set to V(present).
+              This is done after the necessary installation
+              or refresh (upgrade/downgrade) of all the snaps listed in O(name).
+            - See U(https://snapcraft.io/docs/configuration-in-snaps) for more details snap configuration options.
+
         required: false
         type: list
         elements: str
         version_added: 4.4.0
     dangerous:
         description:
-            - Install the given snap file even if there are no pre-acknowledged signatures for it,
-              meaning it was not verified and could be dangerous.
+            - Install the snap in dangerous mode, without validating its assertions and signatures.
+            - This is useful when installing local snaps that are either unsigned or have signatures that haven't been acknowledged.
+            - When installing locally built C(.snap) SquashFS files or downloading them from
+              sources other than the Snap Store, it is necessary to install in dangerous mode
+              with O(dangerous=true) due to the absence of signatures for these snaps.
+            - See U(https://snapcraft.io/docs/install-modes) for more details about installation modes.
         type: bool
         required: false
         default: false
         version_added: 7.2.0
+notes:
+  - Privileged operations, such as installing and configuring snaps, require root priviledges.
+    This is only the case if the user hasn't logged in to the Snap Store.
```

Comment:
I would keep this because a user unaware of the need to escalate will face a very unhelpful message:
`An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible_collections.community.general.plugins.module_utils.mh.exceptions.ModuleHelperException fatal: [noble]: FAILED! => {"changed": false, "channel": null, "classic": false, "cmd": "['state', 'classic', 'channel', 'dangerous', 'avahi']", "msg": "Ooops! Snap installation failed while executing '['state', 'classic', 'channel', 'dangerous', 'avahi']', please examine logs and error output for more details.", "output": {"channel": null, "classic": false, "cmd": "['state', 'classic', 'channel', 'dangerous', 'avahi']", "snap_names": ["avahi"], "snaps_installed": ["avahi"]}, "snap_names": ["avahi"], "snaps_installed": ["avahi"], "vars": {"channel": null, "classic": false, "cmd": "['state', 'classic', 'channel', 'dangerous', 'avahi']", "snap_names": ["avahi"], "snaps_installed": ["avahi"]}}`

Maybe the module can detect the lack of privileges and produce a better message. But in the meantime, I propose keeping the note.

As the current text explains, there are two cases where escalation isn't required:
- using root user for SSH
- not being logged in. Some Ubuntu installations are logged in by default, for example when using the pre-built Ubuntu Core images, the set up wizard performs it.

Reply:
I would keep this because a user unaware of the need to escalate will face a very unhelpful message:
`An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible_collections.community.general.plugins.module_utils.mh.exceptions.ModuleHelperException fatal: [noble]: FAILED! => {"changed": false, "channel": null, "classic": false, "cmd": "['state', 'classic', 'channel', 'dangerous', 'avahi']", "msg": "Ooops! Snap installation failed while executing '['state', 'classic', 'channel', 'dangerous', 'avahi']', please examine logs and error output for more details.", "output": {"channel": null, "classic": false, "cmd": "['state', 'classic', 'channel', 'dangerous', 'avahi']", "snap_names": ["avahi"], "snaps_installed": ["avahi"]}, "snap_names": ["avahi"], "snaps_installed": ["avahi"], "vars": {"channel": null, "classic": false, "cmd": "['state', 'classic', 'channel', 'dangerous', 'avahi']", "snap_names": ["avahi"], "snaps_installed": ["avahi"]}}`

Maybe the module can detect the lack of privileges and produce a better message. But in the meantime, I propose keeping the note.

As the current text explains, there are two cases where escalation isn't required:
- using root user for SSH
- not being logged in. Some Ubuntu installations are logged in by default, for example when using the pre-built Ubuntu Core images, the set up wizard performs it.

Reply:
Ok, fair enough. I would still remove the part that explains how to escalate privilege. If every module that needs it would explain it... it would be an awful amount of repetition.